src/debug/tester/tester.c

```diff
@@ -0,0 +1,275 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Szkudlinski <marcin.szkudlinski@linux.intel.com>
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <rtos/init.h>
+#include <sof/lib/uuid.h>
+#include <sof/audio/component.h>
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/sink_api.h>
+#include <sof/audio/source_api.h>
+#include <sof/audio/sink_source_utils.h>
+#include "tester.h"
+#include "tester_dummy_test.h"
+
+/**
+ * Tester module is a framework for a runtime testing that need a special internal code
+ * i.e. provide an extra load to CPU by running additional thread, etc.
+ *
+ * The idea is to allow testing of special cases using a standard "production" build.
+ * The module should be compiled as a loadable module only, as it is not needed
+ * in any real production case.
+ *
+ * In order to keep only one testing module (in meaning of
+ * a SOF module with separate UUID), a framework is introduced,
+ * where a test to be executed is selected by IPC parameter
+ */
+
+LOG_MODULE_REGISTER(tester, CONFIG_SOF_LOG_LEVEL);
+
+SOF_DEFINE_REG_UUID(tester);
+
+DECLARE_TR_CTX(tester_tr, SOF_UUID(tester_uuid), LOG_LEVEL_INFO);
```

Comment:
since you only want this as a module, maybe add a check that `CONFIG_COMP_TESTER_MODULE` is defined somewhere:
```
#if !CONFIG_COMP_TESTER_MODULE
#error only modular
#endif
```

Reply:
I want it to be modular only for production, but I still prefer to be able to compile it as built-in - it is easier to have one binary in case of debugging.